src/classes/PS_CommitmentRequest.cls

```diff
@@ -286,6 +260,45 @@ public inherited sharing class PS_CommitmentRequest {
         return commitment;
     }
 
+    /***
+    * @description Retrieves Contact data specified as the donor on the Recurring Donation record
+    * that is not yet saved into database
+    * @param contactId Contact Id
+    * @return Contact
+    */
+    private Contact getContact(Id contactId) {
+        List<Contact> contacts = new List<Contact>([
```

Comment:
`new List<Contact>` part is unnecessary because Apex automatically cast the query result into SObject List even if no record is found from the query.
Same with `getAccount()` method below